File: home_assignments/csharp/TicketSeller/services/interfaces/ISuperService.cs
using System;
using System.Collections.Generic;
using model.entities;
using model.validators;

namespace services.interfaces;

public interface ISuperService
{
    void Login(String username, String password,ITicketObserver observer);
    void Logout(String username);
    
    ICollection<Festival> GetAllFestivals();
    
    ICollection<Festival> GetAllFestivalsOnDate(DateTime date);
    
    void SellTicket(int festivalId, String buyerName, int spots);
}

File: home_assignments/java/TicketSeller/TicketSellerServices/src/main/java/ticketseller/services/interfaces/ISuperService.java
package ticketseller.services.interfaces;

import ticketseller.model.entities.Festival;
import ticketseller.services.ServiceException;

import java.time.LocalDate;
import java.util.Collection;

public interface ISuperService {

    void login(String username, String password,ITicketObserver observer) throws ServiceException;

    void logout(String username) throws ServiceException;

    Collection<Festival> getAllFestivals()throws ServiceException;

    Collection<Festival> getAllFestivalsOnDate(LocalDate date) throws ServiceException;

    void sellTicket(Integer festivalID, String buyerName, Integer spots) throws ServiceException;

}
